### ISSUE TITLE
capv controller got stuck from time to time

### ISSUE THREAD

**Issue by jkremser** [giantswarm/roadmap#3019]
context: https://gigantic.slack.com/archives/C01BYMF6RN0/p1701413504028099
cluster: `gcapeverde`

This has happened in past so I am creating an issue for it. Restarting capv controller pod helps to unblock this weird deadlock.

`k tree cluster -n org-t-u4s6cxz t-wx8ohrw`:

![Image](https://github.com/giantswarm/roadmap/assets/535866/0b314885-cd70-4a67-8be1-23a79b69e711)

`k tree vspherecluster -n org-t-u4s6cxz t-wx8ohrw`:

![Image](https://github.com/giantswarm/roadmap/assets/535866/68205662-229e-49ec-bbc7-a852d4d8388b)

**Comment by jkremser** [comment]
logs
- [capv-ctrl-new-logs-after-pod-restart.log](https://github.com/giantswarm/roadmap/files/13526217/capv-ctrl-new-logs-after-pod-restart.log)
- [capv-ctrl-old-before-pod-restart.log](https://github.com/giantswarm/roadmap/files/13526214/capv-ctrl-old-before-pod-restart.log)
- [capi-kubeadm-bootstrap-controller.log](https://github.com/giantswarm/roadmap/files/13526216/capi-kubeadm-bootstrap-controller.log)

**Comment by erkanerol** [comment]
Duplicate of https://github.com/giantswarm/giantswarm/issues/29202
